components/definitions/rewards.d.ts

```diff
@@ -60,20 +62,10 @@ declare namespace Rewards {
     ui: {
       disconnectWalletError: boolean
       modalBackup: boolean
-      modalRedirect:
-          'deviceLimitReachedModal'
+      modalRedirect: ConnectExternalWalletError
```

Comment:
Looks like `modalRedirect` is a union of numbers and strings, which is an odd type. Ideally, we would clean this up somehow, but I can't think of a simple way to do it in this PR.